src/code/z_sram_NES.c

```diff
@@ -1,67 +1,1643 @@
+                        fileChooseCtx->unk_24414[sp76][sp7A] = gSaveContext.save.playerData.playerName[sp7A];
+                    }
+
+                    fileChooseCtx->healthCapacity[sp76] = gSaveContext.save.playerData.healthCapacity;
+                    fileChooseCtx->health[sp76] = gSaveContext.save.playerData.health;
+                    fileChooseCtx->unk_24454[sp76] = gSaveContext.save.inventory.dungeonKeys[9];
+                    fileChooseCtx->unk_24444[sp76] = gSaveContext.save.inventory.questItems;
+                    fileChooseCtx->unk_24458[sp76] = gSaveContext.save.time;
+                    fileChooseCtx->unk_24460[sp76] = gSaveContext.save.day;
+                    fileChooseCtx->unk_24468[sp76] = gSaveContext.save.isOwlSave;
+                    fileChooseCtx->rupees[sp76] = gSaveContext.save.playerData.rupees;
+                    fileChooseCtx->unk_24474[sp76] = CUR_UPG_VALUE(UPG_WALLET);
+
+                    for (sp7A = 0, phi_a0 = 0; sp7A < ARRAY_COUNT(gSaveContext.save.inventory.masks); sp7A++) {
+                        if (gSaveContext.save.inventory.masks[(s32)sp7A] != 0xFF) {
+                            phi_a0++;
+                        }
+                    }
+                    fileChooseCtx->maskCount[sp76] = phi_a0;
+                    fileChooseCtx->heartPieceCount[sp76] =
+                        ((gSaveContext.save.inventory.questItems & 0xF0000000) >> 0x1C);
+                }
+
+                if (sp6E == 1) {
+                    Lib_MemCpy(&(*sramCtx->saveBuf)[0x2000], &gSaveContext.save, sizeof(Save));
+                    func_80146EBC(sramCtx, D_801C67C8[sp64], D_801C6818[sp64]);
+                } else if (!sp6E) { // TODO: == 0?
+                    temp_s2 = gSaveContext.save.checksum;
+                    if (func_80185968(*sramCtx->saveBuf, D_801C67C8[sp64 + 1], D_801C67F0[sp64 + 1])) {
+                        phi_s2_3 = 1;
+                    } else {
+                        Lib_MemCpy(&gSaveContext.save, *sramCtx->saveBuf, sizeof(Save));
+                        phi_s2_3 = gSaveContext.save.checksum;
+                        gSaveContext.save.checksum = 0;
+                        sp7A = Sram_CalcChecksum(&gSaveContext.save, sizeof(Save));
+                    }
+
+                    if (CHECK_NEWF(gSaveContext.save.playerData.newf) || (phi_s2_3 != sp7A) || (phi_s2_3 != temp_s2)) {
+                        func_80185968(*sramCtx->saveBuf, D_801C67C8[sp64], D_801C67F0[sp64]);
+                        Lib_MemCpy(&gSaveContext.save, *sramCtx->saveBuf, sizeof(Save));
+                        Lib_MemCpy(&(*sramCtx->saveBuf)[0x2000], &gSaveContext.save, sizeof(Save));
+                        func_80146EBC(sramCtx, D_801C67C8[sp64], D_801C6818[sp64]);
+                    }
+                }
+            } else if (sp76 < 4) {
+                fileChooseCtx->unk_24468[sp76] = 0;
+
+                if (!CHECK_NEWF(fileChooseCtx->newf2[(s32)sp76])) { // TODO: Needed?
```

Comment:
I think there was talk of reverting `newf2` back to an unknown since we don't really understand it? Or did I miss part of a conversation